grpc/cubic.proto

```diff
@@ -175,119 +148,135 @@ message RecognitionConfig {
   // Unique identifier of the model to use, as obtained from a `Model` message.
   string model_id = 1;
 
-  // The encoding of the audio data to be sent for recognition.
+  // Format of audio data sent/streamed through the `RecognitionAudio` messages.
+  // For formats like WAV/MP3 that have headers, the headers are expected to be
+  // sent at the beginning of the stream, not in every `RecognitionAudio`
+  // message.
+  //
+  // An error is returned if the audio format is not specified.
   //
-  // For best results, the audio source should be captured and transmitted using
-  // the RAW_LINEAR16 encoding.
-  enum Encoding {
-    // Raw (headerless) Uncompressed 16-bit signed little endian samples (linear
-    // PCM), single channel, sampled at the rate expected by the chosen `Model`.
-    RAW_LINEAR16 = 0;
+  // Depending on how they are configured, server instances of this service may
+  // not support all the formats provided in this API.
+  //
+  // For best results, use the RAW audio with Little-Endian SignedInteger PCM
+  // encoding of audio sampled at the same rate as the model you are using.
+  oneof audio_format {
+    AudioFormatRAW audio_format_raw = 2;
+    AudioFormatWithHeaders audio_format_with_headers = 3;
+  }
 
-    // WAV (data with RIFF headers), with data sampled at a rate equal to or
-    // higher than the sample rate expected by the chosen Model.
-    WAV = 1;
+  message AudioFormatRAW {
```

Comment:
This is a minor detail - do we want this message definition to be nested? I suppose it would only matter if we ever needed to use the message type in other places, but I thought it worth checking.

Reply:
Good idea. We could move the enums and this message out to the top level.